### PR TITLE
Fixes to send/retrieve scan data to/from request message

### PR DESCRIPTION
**Pull request by hsong-rh**
Add scan profiles information into request message.

https://bugzilla.redhat.com/show_bug.cgi?id=1533590

### CODE REVIEW

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_vm.rb [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
-        raise "MiqEC2Vm.new: EC2 Instance #{ec2_id} not found" unless ec2_obj
-        raise "MiqEC2Vm.new: Instance of instance store isn't supported!" unless ec2_obj.root_device_type == "ebs"
-        MiqEC2EbsInstance.new(ec2_obj, host_instance, ec2)
       else
         raise "MiqEC2Vm.new: unrecognized ec2 ID #{ec2_id}"
       end
+
+      raise "MiqEC2Vm.new: Instance store isn't supported!" unless ec2_obj&.root_device_type == "ebs"
+
+      ec2_obj.respond_to?(:instance_id) ? MiqEC2EbsInstance.new(ec2_obj, host_instance, ec2, ost) : MiqEC2EbsImage.new(ec2_obj, host_instance, ec2, ost)
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_vm.rb]: This conditional seems strange given we've already performed essentially the same check above.
The refactored `raise unless == "ebs"` no longer indicates if the target is an instance or image.
This no longer indicates the offending `ec2_id` if the object isn't found.

Overall, the code was probably better the way it was.

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_image.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 module AmazonSsaSupport
   class MiqEC2EbsImage < MiqEC2EbsVmBase
-    def initialize(ec2_obj, host_instance, ec2)
+    def initialize(ec2_obj, host_instance, ec2, ost = nil)
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_image.rb]: From the usage, it seems `ost` only contains `scanData`, which isn't used by `MiqVm`.
Given it's also not used by the various `MiqEC2*` classes, why pass it in at all?

Same question applies to all of the following.

File: lib/amazon_ssa_support/ssa_queue_extractor.rb [hunk 3]
```diff
@@ -63,7 +63,10 @@ def do_extract(req)
       @ssaq.delete_request(req)
       extract_reply = @ssaq.new_reply(req)
       begin
-        ec2_vm = MiqEC2Vm.new(req[:ec2_id], @my_instance, @ec2)
+        ost = OpenStruct.new
+        ost.scanData = req[:scan_data]
```
Comment (roliveri) [inline lib/amazon_ssa_support/ssa_queue_extractor.rb]: Do we need `ost` at all?

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_image.rb (hunk 2 above)
Comment (hsong-rh) [reply lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_image.rb]: Wrong. MiqVm uses ost parameter, that's the fix of this PR.

Comment (hsong-rh) [reply lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_image.rb]: https://github.com/hsong-rh/amazon_ssa_support/blob/3a6e25fc432d73dc56877a193b5bc62e7675feab/lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb#L61

This is the place MiqVm uses it. The scan items are set here.

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_vm.rb (hunk 1 above)
Comment (hsong-rh) [reply lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_vm.rb]: The old codes were marked as duplicate by codeclimate checking. I'll change its setting and let it ignore this part.

File: lib/amazon_ssa_support/ssa_queue_extractor.rb (hunk 3 above)
Comment (hsong-rh) [reply lib/amazon_ssa_support/ssa_queue_extractor.rb]: Yes, we do. It contains all scan items we set in analysis profiles.

**Comment by blomquisg** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by blomquisg** [comment]
Fine!

**Pull request merged**